frontend/src/metabase/plugins/index.ts

```diff
@@ -295,11 +296,62 @@ export type IllustrationValue = {
 export const PLUGIN_FORM_WIDGETS: Record<string, ComponentType<any>> = {};
 
 // snippet sidebar
-export const PLUGIN_SNIPPET_SIDEBAR_PLUS_MENU_OPTIONS = [];
+type SnippetSidebarMenuOption = {
+  icon: IconName;
+  name?: string;
+  label?: string;
+  onClick: () => void;
+};
+
+type SnippetSidebarComponentRef = {
```

Comment:
@alxnddr please add separate types for the state, component props, and the plugin itself, and make sure that the components use these types